spacy/tokens/doc.pyx

```diff
@@ -1596,4 +1596,6 @@ def get_entity_info(ent_info):
         ent_type, ent_kb_id, start, end = ent_info
     else:
         ent_id, ent_kb_id, ent_type, start, end = ent_info
+    if isinstance(ent_type, str):
+        ent_type = vocab.strings.add(ent_type)
```

Comment:
@honnibal Is this okay? It feels like there should be a more elegant way but I'm not sure what that would be.

Reply:
You should probably use `hash_string` here and take care of the side-effect of adding to the string store in the calling function.